scene/3d/audio_stream_player_3d.cpp

```diff
@@ -831,7 +832,7 @@ float AudioStreamPlayer3D::get_attenuation_filter_db() const {
 }
 
 void AudioStreamPlayer3D::set_attenuation_model(AttenuationModel p_model) {
-	ERR_FAIL_INDEX(p_model, 3);
+	ERR_FAIL_INDEX(p_model, 4);
```

Comment:
When I set the value to three, in the editor, None appears but I cannot select it. When the value is set to 4, then the build fails (gives warning).

Also it says value is always false, but shouldn't that be the case?

Reply:
Well, maybe just remove the ERR_FAIL_INDEX. It is just saying that something of type AttenuationModel, which is an enum of 4 values, can never be more than or equal to 4.

Reply:
The problem is that this check is useless for the enumeration value, but your can pass an arbitrary int from GDScript. So we still need to check that we're in bounds, no?

For the reference I've seen these warnings a while ago too, but it's weird that they get raised in this PR when the exact same warning was valid before too.

Reply:
I don't know if it is really important but here are the errors raised when set to 3
```
   At: scene/3d/audio_stream_player_3d.cpp:835.
ERROR: set_attenuation_model: Index p_model=3 out of size (3=3)
   At: scene/3d/audio_stream_player_3d.cpp:835.
ERROR: set_attenuation_model: Index p_model=3 out of size (3=3)
   At: scene/3d/audio_stream_player_3d.cpp:835.
ERROR: set_attenuation_model: Index p_model=3 out of size (3=3)
```

Reply:
I suspect if add a cast, then the warning should go:
`if (unlikely((m_index) < 0 || ((int) m_index) >= (m_size)))`

Probably this particular compiler (version) is smart enough to figure out what Bojidar said, and since warnings are forced as errors, it can't compile

Reply:
Sorry I didn't notice your comment.
Would it be better to cast at the macro call like so:
`ERR_FAIL_INDEX((int) p_model, 4);`
or like you proposed in the macro?

Reply:
Well, it's not up to me to decide.
I think adding it in (potentially) every call of this macro is too much hassle.
And I don't think casting here breaks any stuff, because for comparison with a number we are implicitly casting the enum value anyway. Though it might belong in a separate PR :)
As for signed/unsigned, generally signed is better by default, as it helps avoid sneaky bugs with conversion between signed and unsigned.